pkg/policyfilter/state.go

```diff
@@ -105,28 +110,33 @@ func (pod *podInfo) delContainers(id string) []containerInfo {
 
 // containerExists checks returns true if a container exists in the pod
 // it uses log to log inconsistencies in the data.
-func (pod *podInfo) containerExists(log logrus.FieldLogger, containerID string, cgID CgroupID) bool {
+func (pod *podInfo) containerExists(log logrus.FieldLogger, containerID string, cgIDp *CgroupID) bool {
 	for i := range pod.containers {
 		container := &pod.containers[i]
-		if container.id == containerID && container.cgID == cgID {
-			// container was already handled, return
+		if container.id != containerID {
+			continue
+		}
+
+		// we found a matching container id, if no cgroup id is given, return
+		if cgIDp == nil {
 			return true
 		}
 
-		// do some sanity checks and issue a warning if we encounter something strange
-		if container.id == containerID {
-			log.WithFields(logrus.Fields{
-				"container-id":  containerID,
-				"old-cgroup-id": container.cgID,
-				"new-cgroup-id": cgID,
-			}).Warnf("AddPodContainer: conflicting cgroup ids")
-		} else if container.cgID == cgID {
-			log.WithFields(logrus.Fields{
-				"cgroup-id":        cgID,
-				"old-container-id": container.id,
-				"new-container-id": containerID,
-			}).Warnf("AddPodContainer: conflicting container ids")
+		// otherwise, also check that the cgroup id matches
+		if container.id == containerID && container.cgID == *cgIDp {
+			// container was already handled, return
+			return true
 		}
+
+		// issue  a warning and continue
+		// NB: if this happens, we might end up with multiple cgroup
+		// ids for the same container. Since we have no way of knowing
+		// which one is the correct, we keep both.
```

Comment:
Same here under which conditions?

In cgroupv2 it should never happen, in cgroupv1 with multiple hierarchies I think I can see it, but we can definitely track the right hierarchy to reduce such noise

Reply:
> Same here under which conditions?

I'm not aware of any conditions that could cause this.